### PR TITLE
internal: port to new channel-layout API

### PR DESCRIPTION
**Pull request by mbouron**
Makes nope.media compile with FFmpeg 6.1.

### CODE REVIEW

File: .github/workflows/ci_android.yml [hunk 1]
```diff
@@ -42,11 +42,16 @@ jobs:
           compiler="armv7a"
         fi
 
+        rdft_opt="--enable-rdft"
+        if [ ${{ matrix.ffmpeg_version }} == "6.1" ]; then
+            rdft_opt=""
+        fi
+
```
Comment (ubitux) [inline .github/workflows/ci_android.yml]: I believe you can avoid that by extending the matrix like that:
```yaml
      matrix:
        arch: [x86_64, arm, aarch64]
        ffmpeg_version: ["4.4", "6.1"]
        include:
          - ffmpeg_version: "4.4"
            ff_flags: "--enable-rdft"
```

See https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations

File: src/log.c [hunk 2]
```diff
@@ -18,7 +18,9 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <stdio.h>
```
Comment (ubitux) [inline src/log.c]: nit: below `stdarg.h`

File: src/mod_filtering.c [hunk 3]
```diff
@@ -218,10 +218,16 @@ static int setup_filtergraph(struct filtering_ctx *ctx)
         snprintf(args, sizeof(args), "time_base=%d/%d:sample_rate=%d:sample_fmt=%s",
                  time_base.num, time_base.den, codecpar->sample_rate,
                  av_get_sample_fmt_name(codecpar->format));
+#if LIBAVUTIL_VERSION_INT < AV_VERSION_INT(57, 24, 100)
```
Comment (ubitux) [inline src/mod_filtering.c:221]: Ah also please mention the support for recent FFmpeg in the changelog

File: src/log.c (hunk 2 above)
Comment (mbouron) [reply src/log.c]: Fixed.

File: .github/workflows/ci_android.yml (hunk 1 above)
Comment (mbouron) [reply .github/workflows/ci_android.yml]: Done.

File: src/mod_filtering.c (hunk 3 above)
Comment (mbouron) [reply src/mod_filtering.c:221]: Done.

**Review by ubitux — APPROVED** [review]

**Pull request merged**